README.md

```diff
@@ -1 +1,33 @@
-# hershensh
\ No newline at end of file
+# hershensh
+
+## Introduction
+This project aims to build a shell named `hershensh`.
+The goal is for the shell to be fully compliant to [The Open Group Base Specifications, Issue 7, 2018 Edition](https://publications.opengroup.org/standards/unix/c181) which is identical to IEEE Std 1003.1, 2017 Edition.
+
+## Building
+### Requirements
+* C++ compiler supporting C++17.
+* [CMake](https://github.com/Kitware/CMake) version 3.8+.
+
+For running the test suite:
+* [Catch2](https://github.com/catchorg/Catch2) unit testing framework.
+
+### Compiling
+
+Run the commands:
+
+```bash
+mkdir build
+cd build
+cmake ..
+make
+```
+
+This will create the `hershensh` binary.
+
+## Usage
+From the `build` directory run `./hershensh` which will run the hershensh shell.
+
+## Test suite
+
+From the `build` directory run `./runTests`.
```

Comment:
1. I think that this section should come before the usage section as you test before you run the binary.
2. Set up cmake so that we have a test target then you will run the tests with make test

Reply:
1. I think everyone will be running the binary, but not everyone needs to run the tests. So I actually think this order makes more sense.
2. It was already set up, so I guess there are 2 ways to run the tests. I'll update the readme.

Reply:
1. I assumed that the readme was meant for developers. If it's not then you are right.

Reply:
1. It's meant for everyone (including developers).